Bio/PDB/MMCIFParser.py

```diff
@@ -74,8 +75,20 @@ def _mmcif_get(self, key, dict, deflt):
                 return rslt
         return deflt
 
-    def _get_mmcif_header_dict(self, md):
-        dict = {
+    def _update_header_entry(self, target_key, keys, is_float=False):
+        md = self._mmcif_dict
+        for key in keys:
+            val = md.get(key)
+            try:
+                item = val[0]
+            except (TypeError, IndexError):
+                continue
+            if item != "?":
+                self.header[target_key] = item if not is_float else float(item)
```

Comment:
I'm not entirely sure about this line here. The default value will now be `None` for all cases, while the previous version you had (with the default in the function call) was more flexible. The ternary operator call also makes is slower since this comparison is called for every token in the header (right?). I would be more comfortable if this function acted more like a `.get` (like you had before) and then outside we handled the conversion to float explicitly depending on the value. More verbose, but more readable in my opinion.

Reply:
Yes, adding and handling the 'float' option was flakey in retrospect, and better handled outside as you suggest.  The default though is not to update - if the key is not found in the cif dictionary, None is returned and triggers the TypeError in the try/except.  The default is the initialization value in _get_header(), and we only update if there is data that is not "?".